db/dataset.py

```diff
@@ -103,33 +100,30 @@ def update(dataset_id, dictionary, author_id):
     # TODO(roman): Make author_id argument optional (keep old author if None).
     jsonschema.validate(dictionary, BASE_JSON_SCHEMA)
 
-    with create_cursor() as cursor:
+    with db.get_connection() as connection:
         if "description" not in dictionary:
             dictionary["description"] = None
 
-        cursor.execute("""UPDATE dataset
+        connection.execute("""UPDATE dataset
                           SET (name, description, public, author) = (%s, %s, %s, %s)
                           WHERE id = %s""",
                        (dictionary["name"], dictionary["description"], dictionary["public"], author_id, dataset_id))
 
         # Replacing old classes with new ones
-        cursor.execute("""DELETE FROM dataset_class WHERE dataset = %s""", (dataset_id,))
+        connection.execute("""DELETE FROM dataset_class WHERE dataset = %s""", (dataset_id,))
 
         for cls in dictionary["classes"]:
             if "description" not in cls:
                 cls["description"] = None
-            cursor.execute("""INSERT INTO dataset_class (name, description, dataset)
+            result = connection.execute("""INSERT INTO dataset_class (name, description, dataset)
                               VALUES (%s, %s, %s) RETURNING id""",
                            (cls["name"], cls["description"], dataset_id))
-            cls_id = cursor.fetchone()[0]
+            cls_id = result.fetchone()[0]
 
             for recording_mbid in cls["recordings"]:
-                cursor.execute("INSERT INTO dataset_class_member (class, mbid) VALUES (%s, %s)",
+                connection.execute("INSERT INTO dataset_class_member (class, mbid) VALUES (%s, %s)",
                                (cls_id, recording_mbid))
 
-    # If anything bad happens above, it should just rollback by default.
-    commit()
```

Comment:
yes, transactions should automatically commit when `with db.engine.connection()` finishes